File: 1880-check-if-word-equals-summation-of-two-words/1880-check-if-word-equals-summation-of-two-words.cpp
class Solution {
public:
    bool isSumEqual(string firstWord, string secondWord, string targetWord) {
        map<char,int> mp;
        int c = 0;
        for(char i = 'a' ; i <= 'z' ; i++){
            mp[i] = c++;
        }

        string f, s, t;
        for(char c : firstWord) f += to_string(mp[c]);
        for(char c : secondWord) s += to_string(mp[c]);
        for(char c : targetWord) t += to_string(mp[c]);

        while(f.size() > 1 && f[0] == '0') f.erase(0, 1);
        while(s.size() > 1 && s[0] == '0') s.erase(0, 1);
        while(t.size() > 1 && t[0] == '0') t.erase(0, 1);

        int a = stoi(f), b = stoi(s), d = stoi(t);
        return (a + b) == d;
    }
};

File: 1880-check-if-word-equals-summation-of-two-words/1880-check-if-word-equals-summation-of-two-words.py
class Solution:
    def isSumEqual(self, firstWord: str, secondWord: str, targetWord: str) -> bool:
        mp = {}
        c = 0
        for i in range(ord('a'), ord('z') + 1):
            mp[chr(i)] = c
            c += 1

        f, s, t = "", "", ""
        for ch in firstWord:
            f += chr(mp[ch] + ord('0'))
        for ch in secondWord:
            s += chr(mp[ch] + ord('0'))
        for ch in targetWord:
            t += chr(mp[ch] + ord('0'))

        if len(f) > 1 and f[0] == '0':
            f = f[1:]
        if len(s) > 1 and s[0] == '0':
            s = s[1:]
        if len(t) > 1 and t[0] == '0':
            t = t[1:]

        a = int(f)
        b = int(s)
        d = int(t)

        return (a + b) == d
